Commit c2be549 breaks the build

### Is there an existing issue for this?

- [x] I checked and did not find my issue in the already reported ones

### Describe the bug

After commit c2be549 does not build any more for me. Error is:

<...>/darktable/src/lua/lua.h:105:1: error: expected declaration before ‘}’ token
  105 | G_END_DECLS
      | ^~~~~~~~~~~

It looks like the problem is that G_BEGIN_DECLS is inserted inside the `#ifdef USE_LUA`, but `G_END_DECLS` is inserted outside the `#endif`. 

https://github.com/darktable-org/darktable/commit/c2be54933a33fff5a5976d460dc298249daaa89a#diff-4e21dfa7622dc8f782b38fb3aef3d452db34e12b439fa0e4e397ece619d29af2R30-R35

https://github.com/darktable-org/darktable/commit/c2be54933a33fff5a5976d460dc298249daaa89a#diff-4e21dfa7622dc8f782b38fb3aef3d452db34e12b439fa0e4e397ece619d29af2R103-R106

### Steps to reproduce

1. Checkout current master g8fb0f7163c
2. Build it


### Expected behavior

_No response_

### Logfile | Screenshot | Screencast

_No response_

### Commit

_No response_

### Where did you obtain darktable from?

self compiled

### darktable version

release-5.1.0-717-g8fb0f7163c

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 25.04

### Describe your system

_No response_

### Are you using OpenCL GPU in darktable?

None

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

--- a/src/lua/lua.h
+++ b/src/lua/lua.h
@@ -27,11 +27,11 @@
 #include "common/dtpthread.h"
 #include <glib.h>
 
-#ifdef USE_LUA
-
 // while it's poor practice to have includes inside an "extern C" block, we need to
 // do this here because the Lua headers don't supply it when compiling under C++....
 G_BEGIN_DECLS
+
+#ifdef USE_LUA
 
 #include <lautoc.h>
 #include <lauxlib.h>

